### PR TITLE
Logout Feature

### PR DESCRIPTION
**Pull request by Gwillison415**
will require all to add ```PRODUCTION_CALLBACK_URL=""``` 
to their ```.env```

### CODE REVIEW

**Comment by joshwyatt** [comment]
@Gwillison415 please ammend your commit messages, thanks!

File: docker-compose.yml [hunk 1]
```diff
@@ -8,4 +8,4 @@ services:
       - .:/app
     command: npm run dev
     ports:
-      - 8000:8000
+      - 8002:8000
```
Comment (joshwyatt) [inline docker-compose.yml:11]: @Gwillison415 could you fill me in on the need to change this port, thanks!

File: index.js [hunk 2]
```diff
@@ -4,14 +4,31 @@ const graphqlHTTP = require('express-graphql');
...
 const GitHubStrategy = require('passport-github2').Strategy;
+const session = require('express-session');
 const passport = require('passport');
 const partials = require('express-partials');
-const session = require('express-session');
 const cors = require('cors');
 
 const app = express();
-
+/* eslint-disable no-unused-vars */
```
Comment (joshwyatt) [inline index.js:13]: @Gwillison415 please leave me a comment about the need for unused variables.

File: index.js [hunk 3]
```diff
@@ -4,14 +4,31 @@ const graphqlHTTP = require('express-graphql');
...
 const PORT = process.env.PORT || 8000;
+const GITHUB_CLIENT_ID = process.env.GITHUB_CLIENT_ID;
+const GITHUB_CLIENT_SECRET = process.env.GITHUB_CLIENT_SECRET;
+const PRODUCTION_CALLBACK_URL = process.env.PRODUCTION_CALLBACK_URL;
+const LOCAL_CALLBACK_URL = process.env.LOCAL_CALLBACK_URL;
+//
+// const allowCrossDomain = (req, res, next) => {
```
Comment (joshwyatt) [inline index.js]: @Gwillison415 please remove commented code from your codebases before submitting PRs.

File: index.js [hunk 4]
```diff
@@ -53,7 +71,7 @@ passport.deserializeUser((obj, done) => {
 passport.use(new GitHubStrategy({
   clientID: process.env.GITHUB_CLIENT_ID,
   clientSecret: process.env.GITHUB_CLIENT_SECRET,
-  callbackURL: process.env.NODE_ENV === 'development' ? process.env.LOCAL_CALLBACK_URL : 'https:deployed_site.com',
+  callbackURL: process.env.NODE_ENV === 'development' ? process.env.LOCAL_CALLBACK_URL : process.env.PRODUCTION_CALLBACK_URL,
```
Comment (joshwyatt) [inline index.js:62]: @Gwillison415 nice!

File: index.js [hunk 5]
```diff
@@ -85,13 +103,17 @@ app.get('/auth/github/callback',
...
     });
-    res.cookie('isAuth', 'true', {
+    res.cookie('githubAccessToken', process.env.TKN, {
       httpOnly: false,
     });
-
     res.redirect(urlPath());
   },
 );
+app.get('/logout', (req, res) => {
```
Comment (joshwyatt) [inline index.js:100]: @Gwillison415 Nice dude.

**Comment by joshwyatt** [comment]
@Gwillison415 this looks pretty solid, passes all the tests as well as the linter. Please address my comments and ping me back when you have. Thanks so much!

File: docker-compose.yml (hunk 1 above)
Comment (Gwillison415) [reply docker-compose.yml:11]: during mob fix, we decided that the host port for github would map host 8002 to container 8000 in the dashboard. The environmental variables  representing the API endpoints were changed but the docker-compose.yml was not changed. This is what I was hung up on last friday for an hour and a half and we almost solved. This fixed that issue

File: index.js (hunk 2 above)
Comment (Gwillison415) [reply index.js:13]: These variables are not used until the github strategy is implemented. Such that when you lint without being authenticated (i believe), the linter thinks these are unused variables. An alternative solution would be to only lint post-authentication but that seems a touch unrealistic for the Project leads to have to remember

File: docker-compose.yml (hunk 1 above)
Comment (Gwillison415) [reply docker-compose.yml:11]: It seems fitting that the widget would reflect the same mapping as the dashboard

File: index.js (hunk 2 above)
Comment (joshwyatt) [reply index.js:13]: 👍

**Pull request merged**